Fix dropdown casing and config compatibility

vpstackhub opened the pull request:
- Capitalized IO Mode dropdown options: "Write", "Read", "Read & Write"
- Adjusted App.java to support legacy lowercase enum strings using `.toUpperCase()`
- Preserved backwards compatibility with previous properties
- Ensured loadConfig and MainFrame rendering behave correctly

jamesmarkchan commented on src/jdiskmark/MainFrame.java:
```diff
@@ -102,9 +104,9 @@ public void loadConfig() {
-        else if (App.readTest && App.writeTest) { modeStr = "write&read"; }
+        if      (!App.readTest && App.writeTest) { modeStr = "Write"; }
+        else if (App.readTest && !App.writeTest) { modeStr = "Read"; }
+        else if (App.readTest && App.writeTest) { modeStr = "Write & Read"; }
```
Nice work! We can take advantage of your `toString` enhancements with:

```
        Benchmark.IOMode ioMode = Benchmark.IOMode.WRITE;
        if      (!App.readTest && App.writeTest) { ioMode = Benchmark.IOMode.WRITE; }
        else if (App.readTest && !App.writeTest) { ioMode = Benchmark.IOMode.READ; }
        else if (App.readTest && App.writeTest) { ioMode = Benchmark.IOMode.READ_WRITE; }
        else { msg("WARNING: invalid mode detected"); }
        modeCombo.setSelectedItem(ioMode.toString());
```

jamesmarkchan commented:
@vpstackhub this workflow is currently targeting the `master` branch. It would be great if GitHub give you the controls to change the target branch of this pull request to `dev`. When we are preparing the releases we'll create a `release` branch from `dev` and after creating the release from that branch we'll then merge into `master`.

jamesmarkchan commented on src/jdiskmark/MainFrame.java:
```diff
@@ -1157,6 +1161,14 @@ public void adjustSensitivity() {
+    }   
+    // Replace lowercase mode options with proper casing
+
+@SuppressWarnings("unchecked")
```
We should be able to remove this method and just update the initial values of this form based combo box in the design view. If you open the `MainFrame.java` look for the design view on the upper left, select the box and update the model property.

![image](https://github.com/user-attachments/assets/f46443ba-07ac-476f-86b6-ca21c4f5447b)

jamesmarkchan replied:
I think there is even a more advanced way to do this to take advantage of your enumerations. If you select the "..." to the right of the model properties you would trigger this dialog

![image](https://github.com/user-attachments/assets/3f212c2c-a92c-49a9-99dc-fb10938efb23)

And I think we can create a new class that extends from the DefaultComboBoxModel, if so one place to declare it is in `MainFrame.java`.

![image](https://github.com/user-attachments/assets/9947e9fd-f346-487a-b331-495f92ae8b3d)

then can just set the model

![image](https://github.com/user-attachments/assets/aac2774d-036a-4682-86ab-86ff497f1a77)

as well as the selected item

![image](https://github.com/user-attachments/assets/a01657e6-ec0e-459e-8c69-b273f6e3f206)

Not sure what to do about `selectedIndex` property so it's just `-1` in my test right now.

seems to work.

jamesmarkchan commented on src/jdiskmark/MainFrame.java:
```diff
@@ -868,9 +870,11 @@ private void numFilesComboActionPerformed(java.awt.event.ActionEvent evt) {//GEN
-            App.readTest = modeStr.contains("read");
-            App.writeTest = modeStr.contains("write");
+            App.readTest = modeStr.toLowerCase().contains("read");
+            App.writeTest = modeStr.toLowerCase().contains("write");
```
If we you do opt to use the custom model loaded with enums instead of basic strings that the gui builder defaults to this code might need evolve to something like this:
```
        if (modeCombo.hasFocus()) {
            Benchmark.IOMode ioMode = (Benchmark.IOMode) modeCombo.getSelectedItem();
            App.readTest = ioMode == Benchmark.IOMode.READ || ioMode == Benchmark.IOMode.READ_WRITE;
            App.writeTest = ioMode == Benchmark.IOMode.WRITE || ioMode == Benchmark.IOMode.READ_WRITE;
            App.saveConfig();
            System.out.println("modeCombo changed to: " + modeCombo.getSelectedItem());
        }
```

jamesmarkchan commented:
Looking really good @vpstackhub

so to summarize my feedback items:

1. update target branch to be `dev` instead of `master` so we can focus on the changes related to the drop down and filter out all the other changes since dev was branched from master.
2. look at taking advantage of the toString methods you've added by referencing the enums where possible. It is great when the displayed strings are declared a minimum amount of times.
3. remove configureModeCombo() and use basic string initialization in the model attribute or the more advanced custom model approach
4. update README.md and build.xml version property

vpstackhub commented:
Final updates pushed:

- Switched to `IOMode` enum for dropdown (App.java + MainFrame.java)
- Updated README.md with pre-release note
- Updated build.xml with version `dev.dropdown`

Noticed a merge conflict in `build.xml`. Let me know if you'd like me to resolve it or if you'll handle it during review. Ready on my end.

vpstackhub commented:
Confirmed: the UI loads successfully and the IOMode dropdown behaves correctly.
Selecting different modes updates the app state and is reflected in saved config.

jamesmarkchan commented on src/jdiskmark/App.java:
```diff
@@ -252,6 +271,22 @@ public static void saveConfig() {
+if (modeToSave != null) {
+    p.setProperty("ioMode", modeToSave.name());
+}
+        
```
Nice being proactive! I believe we can replace lines 274-289 with a reference to the static:

`p.setProperty("ioMode", ioMode.name());`

If it's not already done, on form load we can also set the static App.ioMode using this logic here.

jamesmarkchan commented on src/jdiskmark/App.java:
```diff
@@ -210,7 +212,24 @@ public static void loadConfig() {
+        App.writeTest = true;
+        break;
+}
+
```
Possible to adjust the white spaces so that the open and close brackets line up?

jamesmarkchan commented on src/jdiskmark/Benchmark.java:
```diff
@@ -34,8 +34,32 @@ public class Benchmark implements Serializable {
+        @Override
+        public String toString() { return "Read & Write"; }
+    }
+}
```
Good to have the close bracket line up with the first column of the corresponding line with the open bracket, in this case the `p` in `public`

jamesmarkchan commented on src/jdiskmark/MainFrame.java:
```diff
@@ -100,17 +109,30 @@ public void loadConfig() {
+}
+
+if (ioMode != null) {
+    modeCombo.setSelectedItem(ioMode);
```
Now that you have introduced the handy new static enum `ioMode` in ln 80 of `App.java`, if we keep that static variables always updated I believe we can replace lines 115-129 with `modeCombo.setSelectedItem(App.ioMode);`

jamesmarkchan commented:
@vpstackhub since you are using the more advanced solution do you want to clear out the configured entries via the NetBeans form controls?

![image](https://github.com/user-attachments/assets/6fa1108e-7920-42c7-8a24-a4d0aeedfcdf)

jamesmarkchan commented:
Hi Valero, just a few items:

1. Whitespace formatting: In `Benchmark.java` lines 38-61 it looks to me the indentation should be increased by one, can highlight these lines and just hit the <tab> button and then save. BTW I know I could quickly do these for you but it would put my name on the commit line and it is more ideal for your name to be associated with these changes in git. In `App.java` lines 188-258 the indentation appears that it should be bumped up by four spaces, again can highlight these lines and just hit the <tab> button. Generally the contents of a function are to be indented (four spaces) from the method declaration and closing bracket and within a class all of it's contents including sub classes and enumerations are indented to indicate it's definition is within a class. Java is a little forgiving in where the spaces are but in languages like python the indentation is very strict. Alternatively the key combination `alt` + `shift` + `F` can be used to tidy up all the indentation in the file.

2. if you could open up `C:\Users\<username>\.jDiskMark\jdm.properties`  with vs code and launch the jdm gui. I am noticing that changing the IO Mode from the drop down in the gui does not change `ioMode=WRITE` in the saved properties files like is done with the `blockSequence=RANDOM` and the "Block Order" dropdown or `palette=BLUE_GREEN` when the "Options > Color Palette" selections are made. ioMode always remains set to WRITE regardless what is selected. I think the read write flags are being set accordingly but probably good for ioMode to be consistent.

    ![image](https://github.com/user-attachments/assets/c1bbec55-db13-4a5b-a6b9-0874dfa42571)

    To resolve In `MainFrame.java` we can set `App.ioMode = mode` after the mode has been read from the combo box
    
    ![image](https://github.com/user-attachments/assets/6aabff4e-a48c-4885-a971-04e7804a742b)

3. ln 876-877 above we can shorten this clause: `mode.name().toLowerCase().contains("read");` to `mode.name().contains("READ");` and `mode.name().toLowerCase().contains("write")` can be `mode.name().contains("WRITE")` since our enumeration names are all in upper case already.

jamesmarkchan commented:
Hey Val, An observation I wanted to share with you is we might be able to deprecated and refactor out the `App.readTest` and `App.writeTest` variables as these can be determined from the new ioMode enumeration state later in the future which would minimize the number of state variables we need to maintain. To do this I believe the exercise would be finding all lines that access those flags and add a helper method that return true or false depending on the active App.ioMode.

vpstackhub commented:
Hey James,
just pushed the final changes removing readTest and writeTest in favor of App.ioMode and the helper methods. Everything has been updated and tested across App.java, BenchmarkWorker, Gui, and MainFrame.

Let me know if you spot anything else I should clean up.
